File: 1046-last-stone-weight/1046-last-stone-weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> pq=new PriorityQueue<>(Collections.reverseOrder());
        for(int i=0;i<stones.length;i++){
            pq.add(stones[i]);
                
        }
        while(pq.size()>1){
            int y=pq.poll();
           
            int x=pq.poll();
    
            
            if(y>x){
              
                y=y-x;
                pq.add(y);
            }
        }
        
        if(pq.size()==1){
            return pq.poll();
        }
        return 0;
    }
}

File: 1046-last-stone-weight/1046-last-stone-weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int> pq;
        for(int i=0;i<stones.size();i++){
            pq.push(stones[i]);
                
        }
        while(pq.size()>1){
            int y=pq.top();
           pq.pop();
            int x=pq.top();
            pq.pop();
    
            
            if(y>x){
              
                y=y-x;
                pq.push(y);
            }
        }
        
        if(pq.size()==1){
            return pq.top();
            
        }
        return 0;
    }
};